internal/integration/cli/jsonpath.go

```diff
@@ -0,0 +1,52 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+//go:build integration_cli
+
+package cli
+
+import (
+	"regexp"
+	"time"
+
+	"github.com/talos-systems/go-retry/retry"
+
+	"github.com/talos-systems/talos/internal/integration/base"
+)
+
+// JSONPathSuite verifies dmesg command.
+type JSONPathSuite struct {
+	base.CLISuite
+}
+
+// SuiteName ...
+func (suite *JSONPathSuite) SuiteName() string {
+	return "cli.JSONPathSuite"
+}
+
+// TestGetScalarPropertyWithJSONPath verifies that the jsonpath filter to the get command can return scalar data.
+func (suite *JSONPathSuite) TestGetScalarPropertyWithJSONPath() {
+	node := suite.RandomDiscoveredNodeInternalIP()
+
+	suite.RunCLI([]string{"get", "--nodes", node, "etcfilestatus", "--output", `jsonpath='{.metadata.namespace}'`},
+		base.StdoutShouldMatch(regexp.MustCompile("files")),
+		base.WithRetry(retry.Constant(15*time.Second, retry.WithUnits(time.Second))),
+	)
+}
+
+// TestGetComplexPropertyWithJSONPath verifies that the jsonpath filter to the get command can return JSON.
+func (suite *JSONPathSuite) TestGetComplexPropertyWithJSONPath() {
+	node := suite.RandomDiscoveredNodeInternalIP()
+
+	const jsonMetadataRegex = `\{\s*"created":\s".*",\s*"id":\s".*",\s*\s*"namespace":\s".*",\s*"owner":\s".*",\s*"phase":\s".*",\s*"type":\s".*",\s*"updated":\s".*",\s*"version":\s\d\n\}`
+
+	suite.RunCLI([]string{"get", "--nodes", node, "etcfilestatus", "--output", `jsonpath='{.metadata}'`},
```

Comment:
can we have something with multiple results? 

I don't know if it's a valid ask, but we could have used `manifests` resource on control plane nodes? It contains a list of items, so something with `[*]` might work there.